src/script_opt/ScriptOpt.cc

```diff
@@ -130,13 +130,19 @@ void optimize_func(ScriptFunc* f, std::shared_ptr<ProfileFunc> pf,
 
 	rc->SetDefSetsMgr(reduced_rds.GetDefSetsMgr());
 
-	auto ud = std::make_unique<UseDefs>(body, rc);
+	auto ud = std::make_shared<UseDefs>(body, rc);
```

Comment:
Why does this become a `shared_ptr` here? You're not passing `ud` anywhere else, so the destruction semantics would remain the same.

Reply:
Turns out that the ZAM compiler will want to keep a copy of it too, hence shared rather than unique.